src/libs/actions/Policy.ts

```diff
@@ -3237,6 +3238,94 @@ function openPolicyDistanceRatesPage(policyID?: string) {
     API.read(READ_COMMANDS.OPEN_POLICY_DISTANCE_RATES_PAGE, params);
 }
 
+function createPolicyDistanceRate(customUnitRate: Rate, customUnitID: string, policyID?: string) {
+    if (!policyID) {
+        return;
+    }
+
+    const optimisticData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                customUnits: {
+                    [customUnitID]: {
+                        rates: {
+                            [customUnitRate.customUnitRateID ?? '']: {
+                                ...customUnitRate,
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const successData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                customUnits: {
+                    [customUnitID]: {
+                        rates: {
+                            [customUnitRate.customUnitRateID ?? '']: {
+                                errors: null,
+                                pendingAction: null,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const failureData: OnyxUpdate[] = [
+        {
+            onyxMethod: Onyx.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.POLICY}${policyID}`,
+            value: {
+                customUnits: {
+                    [customUnitID]: {
+                        rates: {
+                            [customUnitRate.customUnitRateID ?? '']: {
+                                errors: ErrorUtils.getMicroSecondOnyxError('workspace.distanceRates.errors.createRateGenericFailureMessage'),
+                                pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+                            },
+                        },
+                    },
+                },
+            },
+        },
+    ];
+
+    const params: CreatePolicyDistanceRateParams = {
+        policyID,
+        customUnitID,
+        customUnitRate: JSON.stringify(customUnitRate),
+    };
+
+    API.write(WRITE_COMMANDS.CREATE_POLICY_DISTANCE_RATE, params, {optimisticData, successData, failureData});
+}
+
+function clearCreateDistanceRateError(policyID: string, currentRates: Record<string, Rate>, customUnitID?: string, customUnitRateIDToClear?: string) {
```

Comment:
I think this method should be more in line with what we have for categories [here](https://github.com/Expensify/App/blob/cc5f52c944a6072bbda56b4bb94bf7a7e2e85437/src/libs/actions/Policy.ts#L2959-L2971)

```suggestion
function clearDistanceRateError(policyID: string, customUnitID: string, customUnitRateID: string) {
```

Reply:
Hmm, on the second thought maybe it should be split into two. So we have separate methods for clearing errors and removing items from the list. 
There is a case where we display user an error on the list and he can close it. What should happen in such scenario? My idea was that we remove the item together with an error.
![image](https://github.com/Expensify/App/assets/23487612/b2fc75c1-77cf-48c9-89d2-2a5be13be751)

Reply:
I got an answer for this bit from Tom on slack
<img width="869" alt="image" src="https://github.com/Expensify/App/assets/23487612/9ccdb70a-8e90-4f9d-870c-d88619fea527">
I will rename this method so it better described what it does

Reply:
Ok, so I will leave this method as it is for now and once we get to update/delete we can sort this out if needed